### PR TITLE
Documentation: fix building, and minor issues

### PR DESCRIPTION
**Pull request by 1138-4EB**
With the new doc [BUILD.txt](https://github.com/tgingold/ghdl/blob/f345da7692a8417b8249dcf7e0081cd57a283f79/BUILD.txt) is gone, but I didn't realize that the content for GCC and LLVM in the docs are mostly raw copies of the mcode pages:

- http://ghdl.readthedocs.io/en/latest/building/mcode/GNULinux-GNAT.html
- http://ghdl.readthedocs.io/en/latest/building/llvm/GNULinux-GNAT.html
- http://ghdl.readthedocs.io/en/latest/building/gcc/GNULinux-GNAT.html
- http://ghdl.readthedocs.io/en/latest/building/llvm/Windows-MinGW-GNAT.html

This PR fixes `building/gcc/index.rst`, `building/llvm/GNULinux-GNAT` and `building/gcc/GNULinux-GNAT`. It also adds some hints to `building/Building`.

Now, we can safely forget BUILD.txt

Furthermore:

- The links from 'Supported platforms' where pointing to a non-existing anchors inside the page, which made the sidebar menu not to expand properly. It is also fixed for mcode, llvm and gcc.
- A recent bug that prevented some shields from showing properly is also fixed.
- The date of release of 0.35 is added to the News section.
- COPYING.md exists, which includes GPL and CC. Both are raw copies of the source. But GitHub seems not to like it when detecting the License. Then the CC part is moved to COPYING_DOC.md. The non-markdown version must be kept, because GitHub parser does not properly identify markdown variants (benbalter/licensee#247).
- CONTRIBUTING is added, which just recommends the user to visit the corresponding section in the docs.

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,3 @@
+First off, thanks for taking the time to contribute!
+
+Either to [report a bug](http://ghdl-devfork.readthedocs.io/en/doc-gcc/contribute.html#reporting-bugs), [request enhancements](http://ghdl-devfork.readthedocs.io/en/doc-gcc/contribute.html#requesting-enhancements) or to [contribute modifications](http://ghdl-devfork.readthedocs.io/en/doc-gcc/contribute.html#fork-modify-and-pull-request), please read [Contributing](http://ghdl-devfork.readthedocs.io/en/doc-gcc/contribute.html) in the [Docs](http://ghdl-devfork.readthedocs.io/en/doc-gcc/index.html).
```
Comment (tgingold) [inline CONTRIBUTING]: It should be ghdl.readthedocs.io instead of ghdl-devfork

Comment (1138-4EB) [reply CONTRIBUTING]: Sure. Fixed.

**Comment by tgingold** [comment]
thanks.

**Comment by tgingold** [comment]
You must be careful about shields.  If you build ghdl.texi, long URL are not supported.

**Pull request merged**